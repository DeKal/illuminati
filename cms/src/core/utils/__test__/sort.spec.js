import { getComparator, stableSort } from '../sort'

describe('Sort ', () => {
  describe('getComparator ', () => {
    describe('desc ', () => {
      it('should compare more than with no error', function() {
        const comparator = getComparator('desc', 'colA')
        const a = {
          colA: 'a'
        }
        const b = {
          colA: 'b'
        }
        expect(comparator(a, b)).toBe(1)
      })

      it('should compare less than with no error', function() {
        const comparator = getComparator('desc', 'colA')
        const a = {
          colA: 'b'
        }
        const b = {
          colA: 'a'
        }
        expect(comparator(a, b)).toBe(-1)
      })

      it('should compare equal with no error', function() {
        const comparator = getComparator('desc', 'colA')
        const a = {
          colA: 'a'
        }
        const b = {
          colA: 'a'
        }
        expect(comparator(a, b)).toBe(0)
      })
    })
    describe('asc ', () => {
      it('should compare more than with no error', function() {
        const comparator = getComparator('asc', 'colA')
        const a = {
          colA: 'a'
        }
        const b = {
          colA: 'b'
        }
        expect(comparator(a, b)).toBe(-1)
      })

      it('should compare less than with no error', function() {
        const comparator = getComparator('asc', 'colA')
        const a = {
          colA: 'b'
        }
        const b = {
          colA: 'a'
        }
        expect(comparator(a, b)).toBe(1)
      })

      it('should compare equal with no error', function() {
        const comparator = getComparator('asc', 'colA')
        const a = {
          colA: 'a'
        }
        const b = {
          colA: 'a'
        }
        expect(comparator(a, b)).toBe(-0)
      })
    })
  })
  describe('stableSort ', () => {
    it('should sort desc by ColA with no error', function() {
      const array = [
        {
          colA: 'c'
        },
        {
          colA: 'b'
        },
        {
          colA: 'a'
        },
        {
          colA: 'a'
        }
      ]
      expect(stableSort(array, getComparator('desc', 'colA'))).toMatchSnapshot()
    })
    it('should sort asc by ColA with no error', function() {
      const array = [
        {
          colA: 'c'
        },
        {
          colA: 'b'
        },
        {
          colA: 'a'
        }
      ]
      expect(stableSort(array, getComparator('asc', 'colA'))).toMatchSnapshot()
    })
  })
})
