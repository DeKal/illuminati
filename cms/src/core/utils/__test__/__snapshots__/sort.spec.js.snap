// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort  stableSort  should sort asc by ColA with no error 1`] = `
Array [
  Object {
    "colA": "a",
  },
  Object {
    "colA": "b",
  },
  Object {
    "colA": "c",
  },
]
`;

exports[`Sort  stableSort  should sort desc by ColA with no error 1`] = `
Array [
  Object {
    "colA": "c",
  },
  Object {
    "colA": "b",
  },
  Object {
    "colA": "a",
  },
  Object {
    "colA": "a",
  },
]
`;
